bin/opt8n/src/opt8n.rs

```diff
@@ -224,12 +226,27 @@ impl Opt8n {
                 .map(|excess_gas| BlobExcessGasAndPrice::new(excess_gas as u64)),
         };
 
+        let mut env = Env {
+            block: block_env,
+            ..Default::default()
+        };
+        env.cfg.chain_id = self.eth_api.chain_id();
+
+        let handler_cfg = HandlerCfg {
+            spec_id: SpecId::from(self.node_config.hardfork.unwrap_or_default()),
+            // @refcell When we set is_optimism true, execution fails here due to "Failed to load enveloped transaction."
```

Comment:
This seems to be coming from [here](https://github.com/bluealloy/revm/blob/main/crates/revm/src/optimism/handler_register.rs#L207-L212).

It seems like we need to set the enveloped tx in the optimism tx env. Would need to dig into revm more to find where this is set. Maybe it's because we haven't set something somewhere